main.go

```diff
@@ -107,6 +108,7 @@ func main() {
 	scheme := mgr.GetScheme()
 	_ = clusterv1.AddToScheme(scheme)
 	_ = kcpv1.AddToScheme(scheme)
+	_ = corev1.AddToScheme(scheme)
```

Comment:
Is it really necessary to add corev1 to the `scheme`? I believe you added them due to the client that is now getting the secret in order to save the credentials. So as long as you do not watch or modify the `Secret`  object, which I haven't seen in the PR, should we skip adding `corev1` to the `scheme`? WDYT?